protos/feast/core/OnDemandFeatureView.proto

```diff
@@ -41,9 +42,13 @@ message OnDemandFeatureViewSpec {
     // List of features specifications for each feature defined with this feature view.
     repeated FeatureSpecV2 features = 3;
 
-    // List of features specifications for each feature defined with this feature view.
+    // Map of inputs for this feature view.
     map<string, OnDemandInput> inputs = 4;
 
+    // Map of projections for the inputs of this feature view.
+    // Keys must also be present in `inputs`
+    map<string, FeatureViewProjection> projections = 6;
```

Comment:
Can't we add `FeatureViewProjection` in the `OnDemandInput` oneof? And remove the feature view if you so wish?

Reply:
I ended up avoiding this retrieves the entities of the backing FeatureViews.  
https://github.com/feast-dev/feast/blob/7d4369f0b1ce1692bb4fab87c0f70d8e7cdc9fd3/sdk/python/feast/cli.py#L287

This can't be done if a FeatureView Projection is store instead of the FeatureView.

Reply:
Hmmm should we look up the entites expicitly via the feature view name, if it's just in that method?

Reply:
I'd be worried to have two fields in the proto which have this weird dependency between each other, and would prefer a single field - and it feels like we should be able to get away with it.

Reply:
👍 I'll take another pass at doing that tomorrow.

Reply:
This ended up being quite a bit more involved.

The changes in this PR definitely reduce the number of calls to the OnlineStore in the simple case of Features being requested from 'an ODFV which depends on a projection of a FV' and from the unprojected version of that FV (1 call rather than 2 and only the required features).

However, when multiple different Projections are used in a FeatureService I'm not 100% we won't end up making multiple calls to the OnlineStore for each Projection + the projection used by the ODFV.

I'm also not 100% we catch the case where 2 ODFVs used different projections of the same FV. 

But I do think this approach of working out what we need before starting to request things from the OnlineStore is better than it was.